plotly/tools.py

```diff
@@ -1296,3 +1304,686 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
+        quiver = TraceFactory.create_quiver(x, y, u, v)
+
+        # Plot
+        fig=Figure()
+        fig['data'].append(quiver)
+        py.iplot(fig, filename='quiver')
+        ```
+
+        Example 3: Styling the quiver plot
+        ```
+        import numpy as np
+        import math
+
+        # Add data
+        x, y = np.meshgrid(np.arange(-np.pi, math.pi, .5),
+                           np.arange(-math.pi, math.pi, .5))
+        u = np.cos(x)*y
+        v = np.sin(x)*y
+
+        # Create quiver
+        quiver = TraceFactory.create_quiver(x, y, u, v, scale=.2,
+                                            arrow_scale=.3, angle=math.pi/6,
+                                            name='Wind Velocity',
+                                            line=Line(width=1))
+        # Plot
+        fig=Figure()
+        fig['data'].append(quiver)
+        py.iplot(fig, filename='quiver')
+        ```
+        """
+        barb_x, barb_y = _Quiver(x, y, u, v, scale,
+                                 arrow_scale, angle).get_barbs()
+        arrow_x, arrow_y = _Quiver(x, y, u, v, scale,
+                                   arrow_scale, angle).get_quiver_arrows()
+        quiver = Scatter(x=barb_x + arrow_x,
+                         y=barb_y + arrow_y,
+                         mode='lines', **kwargs)
+        return quiver
+
+    @staticmethod
+    def create_streamline(x, y, u, v,
+                          density=1, angle=math.pi / 9,
+                          arrow_scale=.09, **kwargs):
+        """
+        Returns data for a streamline plot.
+
+        :param (list|ndarray) x: 1 dimmensional, evenly spaced list or array
+        :param (list|ndarray) y: 1 dimmensional, evenly spaced list or array
+        :param (ndarray) u: 2 dimmensional array
+        :param (ndarray) v: 2 dimmensional array
+        :param (int) density: controls the density of streamlines in plot.
+            Default = 1
+        :param (angle in radians) angle: angle of arrowhead. Default = pi/9
+        :param (float in [0,1]) arrow_scale: value to scale length of arrowhead
+            Default = .09
+        :param (class) kwargs: kwargs passed through plotly.graph_objs.Scatter
+            for more information on valid kwargs call
+            help(plotly.graph_objs.Scatter)
+
+        :rtype: (trace) returns streamline data
+
+        :raises: (ImportError) will happen if numpy is not installed
+        :raises: (PlotlyError) will happen if x or y are not evenly spaced
+            1 dimmensional lists or arrays
+        :raises: (PlotlyError) will happen if u and v are not the same shape.
+
+        Example 1: Plot simple streamline and increase arrow size
+        ```
+        import numpy as np
+        import math
+
+        # Add data
+        x = np.linspace(-3, 3, 100)
+        y = np.linspace(-3, 3, 100)
+        Y, X = np.meshgrid(x, y)
+        u = -1 - X**2 + Y
+        v = 1 + X - Y**2
+        u = u.T #transpose
+        v = v.T #transpose
```

Comment:
pep:8ball: two spaces before inline comments, oh, just realized this is a docstring, well, still maybe add an extra space : )